_python/scriptmodified.py

```diff
@@ -0,0 +1,95 @@
+from bs4 import BeautifulSoup
+import requests
+import pandas as pd
+import numpy as np
+from sqlalchemy import create_engine
+import numpy as np
+import os
+import logging
+
+
+logging.basicConfig(level=logging.INFO)
+
+class WikiParseWar:
+    def __init__(self,url):
+        self.url = url
+        self.soup = None
+        self.df = pd.DataFrame()
+        logging.info(f'DATABASE_URL: {os.getenv("DATABASE_URL")}')
+    
+
+    def getPage(self):
+        try:
+            page = requests.get(self.url)
+            self.soup = BeautifulSoup(page.text,'html')
+        except requests.RequestException as e:
+            logging.error (f'Error fetching page {e}')
+
+    def getTables(self):
+        tables = self.soup.find_all('table',class_='wikitable')
+        
+        return tables
+    
+    def parseTables(self,table_index):
+        tables = self.getTables()
+        rus_fed_table = tables[table_index]
+
+        headers = [header.get_text(strip=True) for header in rus_fed_table.find_all('th')]
+        logging.info(f"Parsed headers: {headers}")
+
+        df = pd.DataFrame(columns=headers)
+
+        column_data = rus_fed_table.find_all('tr')
+        for row in column_data[1:]:
+            row_data = row.find_all('td')
+            each_row = [data.text.strip() for data in row_data]
+            #print(each_row)
+
+            length = len(df)
+            df.loc[length] = each_row
```

Comment:
This is inefficient, please use standard python types and then make the type a DataFrame